pvHelpers/crypto/asymm_key/asymm_key_v3.py

```diff
@@ -83,7 +92,20 @@ def public_key(self):
 
     @params(object, bytes)
     def unseal(self, cipher):
-        return FC.hybrid_unseal(self._curve25519_secret, self._p256_secret, cipher)
+        try:
+            buffer = SealedDataBuffer()
+            buffer.ParseFromString(cipher)
+
+            if buffer.proof != 1:
+                return FC.hybrid_unseal(self._curve25519_secret, self._p256_secret, cipher, False)
```

Comment:
shouldn't we raise the error here?
If proof exists, shouldn't it always be 1? otherwise, doesn't this mean the buffer is corrupted?

Reply:
So the case where the buffer parses but the proof isn't 1 is basically a coincidental successful parsing of a raw ciphertext, which is why we fallback to the legacy method here.